tidy3d/components/simulation.py

```diff
@@ -2052,3 +2052,76 @@ def custom_datasets(self) -> List[Dataset]:
         """
         datasets = [src.field_dataset for src in self.sources if isinstance(src, CustomFieldSource)]
         return datasets
+
+    def volumetric_equivalent(self) -> Simulation:
+        """Generate new Simulation wherein any 2D materials are converted to 3D
+        volumetric equivalents."""
+
+        small_num = 1e-5
+
+        # get grid size at monolayer
+        def get_dl(sim, geom, axis):
+            dls = sim.discretize(geom).sizes.to_list[axis]
+            if len(dls) != 1:
+                log.error("2d materials do not support this nonuniform grid.")
+            return dls[0]
+
+        # snap monolayer to Yee grid
```

Comment:
same commend about type hints, sim arg, and Docstring. Also in the function below.